### ISSUE TITLE
Token for application rpc should support IP service name

### ISSUE THREAD

**Issue by yuanzac** [linkedin/TonY#284]
The token generated in ApplicationMaster.prepare() uses hostname as token's service:

```
// create token for application RPC server
  Token<? extends TokenIdentifier> tensorflowClusterToken = new Token<>(identifier, secretManager);
  tensorflowClusterToken.setService(new Text(amHostPort));
  UserGroupInformation.getCurrentUser().addToken(tensorflowClusterToken);
```

The token's service name should be specified with the configuration of hadoop.security.token.service.use_ip

**Comment by erwa** [comment]
Interesting, in our cluster, we have `hadoop.security.token.service.use_ip=false`, which is probably why I haven't seen this error before.

**Comment by oliverhu** [comment]
All tests also pass locally for me 🤔

**Comment by oliverhu** [comment]
fixed in #298
